internal/dbadapter/db_adapter.go

```diff
@@ -71,6 +72,29 @@ type JobInsertResult struct {
 	UniqueSkippedAsDuplicate bool
 }
 
+type SortOrder int
+
+const (
+	SortOrderUnspecified SortOrder = iota
+	SortOrderAsc
+	SortOrderDesc
+)
+
+type JobListOrderBy struct {
+	Expr  string
+	Order SortOrder
+}
+
+type JobListParams struct {
+	Conditions string
```

Comment:
Just checking here: conditions are meant to be strictly stay an internal construct right? Just wondering if we have to do anything about making sure that SQL injections aren't possible.

Reply:
Yes indeed. The thought is we'll only expose it in ways that do not accept raw user input. Obviously it would be ideal if we had a fully flexible query builder to do this all in a safe way, but barring that we'll just need to be mindful when expanding use of this internal API to ensure we don't add any SQL injection opportunities. Or if we do that would have to be very clearly called out as dangerous, but hopefully we won't find a reason to need to do that.